libraries/AP_RangeFinder/RangeFinder.cpp

```diff
@@ -554,6 +562,13 @@ RangeFinder::RangeFinder_Status RangeFinder::status(uint8_t instance) const
     return state[instance].status;
 }
 
+void RangeFinder::handle_msg(mavlink_message_t *msg) {
```

Comment:
I wouldn't bother with the check of the RangeFinder type.. I'd just make sure that instance isn't null and then pass it in and most times it'll just end up using the default implementation and returning immediately.

Using the instance id from the message and making sure it's the same as the instance id of the driver makes sense if we want to use multiple range finders (all supplied from the companion computer).. but because nobody wants that functionality immediately I think it's fine as it is.